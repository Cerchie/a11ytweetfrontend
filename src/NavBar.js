import * as React from 'react'
import { useContext, useRef } from 'react'
import { Link } from 'react-router-dom'
import UserContext from './UserContext'

function NavBar({ logout }) {
    //so we can focus for a11y
    const linkInput = useRef(null)

    //using the ref to focus the element
    function handleRender(linkInput) {
        linkInput.current.focus()
    }

    if (useContext(UserContext) !== undefined) {
        const { currentUser } = useContext(UserContext)

        function loggedOutNav() {
            return (
                <nav
                    role="navigation"
                    aria-label="Nav"
                    className="rounded bg-blue-50 text-red-600 text-2xl p-6 m-6"
                >
                    <ul>
                        <li ref={linkInput}>
                            <Link to="/login">Login</Link>
                        </li>
                        <li>
                            <Link to="/signup">Signup</Link>
                        </li>
                        <li>
                            <Link to="/">Home</Link>
                        </li>
                        <li>
                            <Link to="/links">a11y repos</Link>
                        </li>
                    </ul>
                </nav>
            )
        }

        function loggedInNav() {
            return (
                <>
                    <h1 className="rounded bg-blue-50 text-red-600 text-2xl p-6 m-6">
                        <Link exact to="/" className="navbar-brand">
                            Hi {currentUser.username}!
                        </Link>
                    </h1>
                    <nav
                        role="navigation"
                        aria-label="Nav"
                        className="rounded bg-blue-50 text-red-600 text-2xl p-6 m-6"
                    >
                        <ul>
                            <li>
                                <Link to="/profile">Edit Profile</Link>
                            </li>
                            <li>
                                <Link to="/userlinks">Your Links</Link>
                            </li>
                            <li>
                                <Link to="/links">
                                    List of a11y-related links
                                </Link>
                            </li>
                            <li>
                                <Link
                                    className="nav-link"
                                    to="/"
                                    onClick={logout}
                                >
                                    Log out{' '}
                                    {currentUser.first_name ||
                                        currentUser.username}
                                </Link>
                            </li>
                        </ul>
                    </nav>
                </>
            )
        }

        return (
            <nav className="Navigation navbar navbar-expand-md">
                {currentUser ? loggedInNav() : loggedOutNav()}
            </nav>
        )
    }
    return <> No navbar. </>
}

export default NavBar

//code adapted from my Springboard projecthttps://github.com/Cerchie/react-jobly/blob/main/frontend/src/NavBar.js
