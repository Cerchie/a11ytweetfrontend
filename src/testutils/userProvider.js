import React from 'react'
import UserContext from '../UserContext'

const demoUser = {
    username: 'testuser',
}

const UserProvider = ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
        {children}
    </UserContext.Provider>
)

export { UserProvider }
