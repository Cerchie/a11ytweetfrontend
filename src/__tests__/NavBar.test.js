import { render, screen } from '@testing-library/react'
import React from 'react'
import NavBar from '../NavBar'
import { BrowserRouter } from 'react-router-dom'
import { MemoryRouter } from 'react-router'
import { UserProvider } from '../testutils/userProvider'
test('renders comp', () => {
    render(
        <BrowserRouter>
            <NavBar />
        </BrowserRouter>
    )
})

it('matches snapshot', function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <NavBar />
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
})

it('matches snapshot when logged out', function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider currentUser={null}>
                <NavBar />
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
})
//not worried about the 'received true...' error for this reason https://github.com/styled-components/styled-components/issues/1198
