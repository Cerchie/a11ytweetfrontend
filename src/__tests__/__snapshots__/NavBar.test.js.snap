// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar navbar-expand-md"
  >
    <h1>
      <a
        class="navbar-brand"
        href="/"
      >
        Welcome testuser!
      </a>
    </h1>
    <nav
      aria-label="Nav"
      role="navigation"
    >
      <ul>
        <li>
          <a
            href="/profile"
          >
            Edit Profile
          </a>
        </li>
        <li>
          <a
            href="/userlinks"
          >
            Your Links
          </a>
        </li>
        <li>
          <a
            href="/links"
          >
            List of a11y-related links
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/"
          >
            Log out testuser
          </a>
        </li>
      </ul>
    </nav>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar navbar-expand-md"
  >
    <nav
      aria-label="Nav"
      class="rounded bg-blue-50 text-red-600 text-2xl p-6 m-6"
      role="navigation"
    >
      <ul>
        <li>
          <a
            href="/login"
          >
            Login
          </a>
        </li>
        <li>
          <a
            href="/signup"
          >
            Signup
          </a>
        </li>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/links"
          >
            a11y repos
          </a>
        </li>
      </ul>
    </nav>
  </nav>
</DocumentFragment>
`;
